.gitmodules

```diff
@@ -34,7 +34,10 @@
   path = DATM
   url = https://github.com/NOAA-EMC/NEMSdatm
   branch = develop
-[submodule "CDEPS"]
+[submodule "GOCART"]
```

Comment:
When I suggested the same as you do here to @aerorahul , I was advised that it would be better to put the submodule in `GOCART-interface/GOCART` and have a `CMakeLists.txt` file in `GOCART-interface` that can control any logic that is specific to `GOCART`. See https://github.com/noaa-gsl/ufs-weather-model/tree/gsl/develop-chem for the implementation.

Reply:
I would still recommend doing that.
I see including "GOCART" as is currently done is also bringing in `ecbuild` and `esma-cmake` dependencies.
By doing that, we basically are bringing in ECMWF's build and GMAO's adaptations to ECMWF's build into the UFS.
Everyone wants to write their own wrappers to CMake. This makes swapping and sharing extremely difficult, prone to unparalleled entropy increase and a nightmare for maintenance.  
GOCART is the aerosol component.  We should not need to bring it its build framework.
I recommend dropping the ecbuild and esma-cmake dependencies.

Reply:
The UFS Aerosols component was developed collaboratively by NASA and NOAA so that it could be shared between the UFS and GEOS-5. This also included the CMake build framework, which needs to work for both models and reside in the authoritative GOCART repository at NASA. Adding a new `CMakeLists.txt` in `GOCART-interface` defeats the original mandate, placing the burden of maintaining the GOCART build system on the UFS.

Work is ongoing with NASA to minimize the number of dependencies required to build UFS-Aerosols outside GEOS-5.
While I agree that `esma_cmake` functions could be replaced with plain CMake commands, this set of macros is heavily used by NASA and therefore cannot be removed at this time. The `ecbuild` dependency is also required by JEDI.

Reply:
I am not against GOCART.
The CMake build framework was *not* developed collaboratively and is not meant to be shared between the frameworks.
I am advocating against the build system taken without question from GEOS.
Does implementing a clean cmake in GOCART-interface change the science of aerosols?
If it does, that is a problem that needs addressing.

ecbuild is used in JEDI and is not an approved software to be installed on WCOSS yet.  Are we going to wait till it and other dependencies that are coming with this PR to be installed on WCOSS? How is testing going to proceed with Aerosols on WCOSS?

Adding ecbuild and esma-cmake adds burden on the UFS, that is clearly unnecessary.

Adding components to the UFS should not mean that the UFS has to buy every component's native build framework.  The UFS should be able to build the science component.

Reply:
@aerorahul - I share your view on this and I'd like to clarify a few points:
- We are not buying the build system from GEOS without question. The current GOCART build system is indeed the result of collaborative efforts at NASA and NOAA that have added in-depth changes to its original version over time to adapt to UFS. These changes now allow to install GOCART dependencies as individual system libraries (as required by EMC), make them available via the Modules package, and use them when building GOCART via standard CMake calls to `find_package()`. Work is also ongoing to minimize the number of dependencies required by GOCART.
The goal is to keep the GOCART build system in the authoritative repository, so that it can be maintained and used collaboratively. Any changes to the build system (including CCPP-related extensions) should be committed back to this repository. While I'd be happy to discuss with GMAO how to continue to streamline the GOCART build system, adding a separate build system looks like a duplication of efforts and does not seem to fulfill the deliverable set forth in the UFS R2O project.
- P7 is the initial target for GOCART integration. Initial model evaluations with prognostic aerosols will be mainly performed on Hera. We will need to find an agreed upon solution for `ecbuild` and other packages while P7 evaluation runs are being performed.
- The `esma_cmake` package is maintained by GMAO (https://github.com/GEOS-ESM/ESMA_cmake). Any issue with this package will be directed to GEOS developers, not UFS.

Reply:
In my opinion, the GOCART build system should be made generic and without dependencies of ecbuild and esma-cmake for use in the community and UFS.
It allows for full control of the GOCART build and removes excessive dependencies.
I am very familiar on how cmake works and will gladly assist in creating a GOCART-interface that can build GOCART as desired in the UFS.

Reply:
In the end, we should ask the following questions:
- is this making UFS better in the long term
- is this going to help us maintain the UFS in the long term

Do people think about this or just want to get their stuff into `develop` and move on?

Reply:
Internal discussions resulted in changes that resolved this conversation. See, for instance, commit 35d7f86f8d52b4c57301633d224678af9dd60945, which removed both `ecBuild` and `ESMA_cmake` dependencies, and a3f15c4cc9724c26963a398e771730f1eb96e6d2, which eliminated the need for `GOCART-interface`.